src/Compilers/CSharp/Portable/Binder/Imports.cs

```diff
@@ -576,6 +576,18 @@ private void Validate()
                 alias.Alias.CheckConstraints(semanticDiagnostics);
             }
 
+            foreach (var @using in Usings)
+            {
+                // Check if `using static` directives meet constraints.
+                if (@using.NamespaceOrType.IsType)
+                {
+                    var typeSymbol = (TypeSymbol)@using.NamespaceOrType;
+                    var corLibrary = typeSymbol.ContainingAssembly.CorLibrary;
+                    var conversions = new TypeConversions(corLibrary);
+                    typeSymbol.CheckAllConstraints(conversions, @using.UsingDirective.Name.Location, semanticDiagnostics);
```

Comment:
Looks like there's an NRE in here.

Reply:
It looks like UsingDirective can be null in scripting scenarios? Do I need to supply an alternate location in such cases?

Reply:
> It looks like UsingDirective can be null in scripting scenarios? Do I need to supply an alternate location in such cases?

We usually use ```NoLocation.Singleton``` in situations when there is no specific syntax to refer to.

---
In reply to: [227996503](https://github.com/dotnet/roslyn/pull/30737#discussion_r227996503) [](ancestors = 227996503)